src/typechecker.c

```diff
@@ -890,7 +889,35 @@ NODISCARD bool typecheck_expression(AST *ast, Node *expr) {
     } break;
 
     /// Make sure a cast is even possible.
-    case NODE_CAST: TODO();
+    case NODE_CAST: {
+      Type *t_to = expr->type;
+      if (type_is_incomplete(t_to))
+        ERR(t_to->source_location, "Can not cast to incomplete type %T", t_to);
+
+      if (!typecheck_expression(ast, expr->cast.value))
+        return false;
+
+      Type *t_from = expr->cast.value->type;
+
+      // TODO: What do we do when from and to types are equal?
+
+      if (type_is_incomplete(t_from))
+        ERR(expr->cast.value->source_location, "Can not cast from an incomplete type %T", t_from);
```

Comment:
This is technically not true since we *can* cast from the same incomplete type to itself (e.g. from `void` to `void`).

Reply:
See also `common_type()` and `convertible()` or whatever they’re called for how that case is handled there.